File: other-languages/token/token.rb
# -*- coding: UTF-8 -*-

=begin
Token

Autor:
    ?
Colaborador:
    Bruno Lara Tavares <bruno.exz@gmail.com>
    Felipe Djinn <felipe@felipedjinn.com.br>
Tipo:
    sequence
Descrição:
    Gera um token aleatório
Complexidade:
    ?
Dificuldade:
    facil
=end

def token(length = 10)
    rand(36**length).to_s 36
end


"""
Examples
"""

puts "Token com 10 caracteres (padrão): " + token()
puts "Token com 5 caracteres: " +token(5)
puts "Token com 15 caracteres " +token(15)

File: other-languages/sequence/token/token.php
<?php

/*
Token
Autor:
    Sérgio Mandrake <sergio_mandrake@yahoo.com.br>
Colaborador:
    Sérgio Mandrake <sergio_mandrake@yahoo.com.br>
Tipo:
    sequence
Descrição:
    Gera um token aleatório
Complexidade:
    ?
Dificuldade:
    facil
*/

function token($comprimento = 6) {
    $caracteres = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
    $totalCarac = strlen($caracteres) -1;
    $resultado = '';
    for ($i = 0; $i < $comprimento; $i++) {
        $resultado .= $caracteres[mt_rand(0, $totalCarac)];
    }

    return $resultado;
}

echo token(5);
